### PR TITLE
KMS setup for vault auto-unseal in AWS

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/5436

### CODE REVIEW

**Comment by teemow** [comment]
the changes lgtm. but can you explain a bit this will look in the end? is kms used within the same account the installation is in or a different?

File: modules/aws/vault/vault-iam.tf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:DescribeKey"
+    ]
+  }
+}
+
+resource "aws_iam_role_policy" "vault-ksm-unseal" {
```
Comment (mazzy89) [inline modules/aws/vault/vault-iam.tf]: Typo

File: modules/aws/vault/vault-iam.tf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  name = "${var.cluster_name}-vault"
+  role = "${aws_iam_role.vault.name}"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  # Sleep a little to wait the IAM profile to be ready -
+  provisioner "local-exec" {
+    command = "sleep 10"
```
Comment (mazzy89) [inline modules/aws/vault/vault-iam.tf]: Look very hackish. Which is the reason for that? The reference is pretty explicit in the `aws_instance.vault` in the line

```hcl
  iam_instance_profile = "${aws_iam_instance_profile.vault.id}"
```

so Terraform will care about that.

**Comment by corest** [comment]
@teemow it is used within the same account, where CP is installed.

File: modules/aws/vault/vault-iam.tf (hunk 2 above)
Comment (corest) [reply modules/aws/vault/vault-iam.tf]: copy past from worker

**Pull request merged**